include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -0,0 +1,259 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file range/detail/random_access_iterator.hpp
+ * \brief Random access iterator for const and non const containers.
+ * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+ * \ingroup iterator
+ */
+
+#pragma once
+
+#include <type_traits>
+#include <cassert>
+
+#include <range/v3/utility/iterator_traits.hpp>
+
+#include <seqan3/core/concept/iterator.hpp>
+#include <seqan3/range/concept.hpp>
+
+namespace seqan3::detail
+{
+
+/*! \tparam container_type {Datastructure on which the iterator operates, e.g. std::vector<int>.}
+ *  \class random_access_iterator
+ *  \brief Implementation of a random access iterator on an input container.
+ *  \details No iterator operation will modify the container. Arithmetic and boolean operations
+ *  are applied to the iterator positions, not the corresponding values of their
+ *  containers.
+ */
+template <typename container_type>
+    requires random_access_range_concept<container_type> && sized_range_concept<container_type>
+class random_access_iterator
+{
+
+private:
+    //! \brief Iterator stores pointer to underlying container structure
+    typename std::add_pointer_t<container_type> host{nullptr};
+    //! \brief Use container's size_type as a position
+    using position_type =  ranges::v3::size_type_t<container_type>;
+    //! \brief Store position index for container
+    position_type pos{static_cast<position_type>(0)};
+
+public:
+    //! \brief Type for distances between iterators
+    using difference_type = ranges::v3::difference_type_t<container_type>;
+    //! \brief Value type of container elements
+    using value_type = ranges::v3::value_type_t<container_type>;
+    //! \brief Use reference type defined by container
+    using reference = typename container_type::reference;
+    //! \brief Use const reference type provided by container
+    using const_reference = typename container_type::const_reference;
+    //! \brief Pointer type is pointer of container element type
+    using pointer = value_type *;
+    //! \brief Tag this class as a random access iterator
+    using iterator_category = std::random_access_iterator_tag;
+
+    /*!\name Constructors/Destructors
+     * \{
+    */
+    // \brief Default constructor
+    constexpr random_access_iterator() = default;
+
+    //! \brief Construct by host, default position pointer with 0
+    constexpr random_access_iterator(container_type & host) : host{&host} {}
+
+    //! \brief Construct by host and explicit position
+    constexpr random_access_iterator(container_type & host, const position_type pos) : host{&host}, pos{pos} {}
```

Comment:
the `const` comes after the type according to our guidelines, also in the other places!